ToolBox/KaikoDataConverter/KaikoCryptoReader.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Ionic.Zlib;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using ZipEntry = Ionic.Zip.ZipEntry;
+
+namespace QuantConnect.ToolBox.KaikoDataConverter
+{
+    /// <summary>
+    /// Decompress single entry from Kaiko crypto raw data.
+    /// </summary>
+    public class KaikoCryptoreader
+    {
+        /// <summary>
+        /// Gets the raw data from entry.
+        /// </summary>
+        /// <param name="zipEntry">The zip entry.</param>
+        /// <returns>IEnumerable with the zip entry content.</returns>
+        public static IEnumerable<string> GetRawDataFromEntry(ZipEntry zipEntry)
+        {
+            using (var outerStream = new StreamReader(zipEntry.OpenReader()))
+            using (var innerStream = new GZipStream(outerStream.BaseStream, CompressionMode.Decompress))
+            {
+                var decompressed = Decompres(innerStream);
+                return Encoding.ASCII.GetString(decompressed).Split(new char[] { '\n' });
```

Comment:
I imagine these files aren't up big, so no issue, but fyi, split can be dangerous with large files. [Here's](https://www.codeproject.com/Articles/176031/Out-of-Memory-Exception-A-simple-string-Split-can) a lengthy bit interesting read on the mayhem that can ensue.

Reply:
Thank you very much for the link. 

Before I implemented this, I checked files sizes. Those are tick files for a single day, for a single symbol. The biggest is a 37 MB quote file for 2017-12-16.